### ISSUE TITLE
"Data Ingestion - Intake" notebook triggers SSL error

### ISSUE THREAD

**Issue by tylere** [ProjectPythia/landsat-ml-cookbook#10]
**What happened**:

Code cell triggered error:
```AttributeError: module 'lib' has no attribute 'OpenSSL_add_all_algorithms'```

**What you expected to happen**:

The code cell should execute without an error.

**Minimal Complete Verifiable Example**:

1. Open https://projectpythia.org/landsat-ml-cookbook/notebooks/1.1_Data_Ingestion-Intake.html
2. Click "Launch Binder"
3. Menu -> Run -> Run All Cells

**Environment**:

Environment provided by Binder.

**Stack Trace**
```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Cell In[34], line 1
----> 1 landsat_5 = cat.landsat_5_small

File /srv/conda/envs/notebook/lib/python3.10/site-packages/intake/catalog/base.py:391, in Catalog.__getattr__(self, item)
    388 if not item.startswith('_'):
    389     # Fall back to __getitem__.
    390     try:
--> 391         return self[item]  # triggers reload_on_change
    392     except KeyError as e:
    393         raise AttributeError(item) from e

File /srv/conda/envs/notebook/lib/python3.10/site-packages/intake/catalog/base.py:436, in Catalog.__getitem__(self, key)
    427 """Return a catalog entry by name.
    428 
    429 Can also use attribute syntax, like ``cat.entry_name``, or
   (...)
    432 cat['name1', 'name2']
    433 """
    434 if not isinstance(key, list) and key in self:
    435     # triggers reload_on_change
--> 436     s = self._get_entry(key)
    437     if s.container == 'catalog':
    438         s.name = key

File /srv/conda/envs/notebook/lib/python3.10/site-packages/intake/catalog/utils.py:45, in reload_on_change.<locals>.wrapper(self, *args, **kwargs)
     42 @functools.wraps(f)
     43 def wrapper(self, *args, **kwargs):
     44     self.reload()
---> 45     return f(self, *args, **kwargs)

File /srv/conda/envs/notebook/lib/python3.10/site-packages/intake/catalog/base.py:323, in Catalog._get_entry(self, name)
    321 ups = [up for name, up in self.user_parameters.items() if name not in up_names]
    322 entry._user_parameters = ups + (entry._user_parameters or [])
--> 323 return entry()

File /srv/conda/envs/notebook/lib/python3.10/site-packages/intake/catalog/entry.py:77, in CatalogEntry.__call__(self, persist, **kwargs)
     75     raise ValueError('Persist value (%s) not understood' % persist)
     76 persist = persist or self._pmode
---> 77 s = self.get(**kwargs)
     78 if persist != 'never' and isinstance(s, PersistMixin) and s.has_been_persisted:
     79     from ..container.persist import store

File /srv/conda/envs/notebook/lib/python3.10/site-packages/intake/catalog/local.py:290, in LocalCatalogEntry.get(self, **user_parameters)
    287     return self._default_source
    289 plugin, open_args = self._create_open_args(user_parameters)
--> 290 data_source = plugin(**open_args)
    291 data_source.catalog_object = self._catalog
    292 data_source.name = self.name

File /srv/conda/envs/notebook/lib/python3.10/site-packages/intake_xarray/raster.py:57, in RasterIOSource.__init__(self, urlpath, chunks, concat_dim, xarray_kwargs, metadata, path_as_pattern, storage_options, **kwargs)
     55     self._can_be_local = fsspec.utils.can_be_local(self.urlpath[0])
     56 else:
---> 57     self._can_be_local = fsspec.utils.can_be_local(self.urlpath)
     58 super(RasterIOSource, self).__init__(metadata=metadata)

File /srv/conda/envs/notebook/lib/python3.10/site-packages/fsspec/utils.py:406, in can_be_local(path)
    403 from fsspec import get_filesystem_class
    405 try:
--> 406     return getattr(get_filesystem_class(get_protocol(path)), "local_file", False)
    407 except (ValueError, ImportError):
    408     # not in registry or import failed
    409     return False

File /srv/conda/envs/notebook/lib/python3.10/site-packages/fsspec/registry.py:211, in get_filesystem_class(protocol)
    209 bit = known_implementations[protocol]
    210 try:
--> 211     register_implementation(protocol, _import_class(bit["class"]))
    212 except ImportError as e:
    213     raise ImportError(bit["err"]) from e

File /srv/conda/envs/notebook/lib/python3.10/site-packages/fsspec/registry.py:234, in _import_class(cls, minv)
    232 else:
    233     mod, name = cls.rsplit(".", 1)
--> 234     mod = importlib.import_module(mod)
    235     return getattr(mod, name)

File /srv/conda/envs/notebook/lib/python3.10/importlib/__init__.py:126, in import_module(name, package)
    124             break
    125         level += 1
--> 126 return _bootstrap._gcd_import(name[level:], package, level)

File <frozen importlib._bootstrap>:1050, in _gcd_import(name, package, level)

File <frozen importlib._bootstrap>:1027, in _find_and_load(name, import_)

File <frozen importlib._bootstrap>:1006, in _find_and_load_unlocked(name, import_)

File <frozen importlib._bootstrap>:688, in _load_unlocked(spec)

File <frozen importlib._bootstrap_external>:883, in exec_module(self, module)

File <frozen importlib._bootstrap>:241, in _call_with_frames_removed(f, *args, **kwds)

File /srv/conda/envs/notebook/lib/python3.10/site-packages/s3fs/__init__.py:1
----> 1 from .core import S3FileSystem, S3File
      2 from .mapping import S3Map
      4 from ._version import get_versions

File /srv/conda/envs/notebook/lib/python3.10/site-packages/s3fs/core.py:28
     26 import aiobotocore
     27 import botocore
---> 28 import aiobotocore.session
     29 from aiobotocore.config import AioConfig
     30 from botocore.exceptions import ClientError, HTTPClientError, ParamValidationError

File /srv/conda/envs/notebook/lib/python3.10/site-packages/aiobotocore/session.py:1
----> 1 from botocore import UNSIGNED, translate
      2 from botocore.exceptions import PartialCredentialsError
      3 from botocore.session import (
      4     EVENT_ALIASES,
      5     ServiceModel,
   (...)
      8     copy,
      9 )

File /srv/conda/envs/notebook/lib/python3.10/site-packages/botocore/translate.py:16
      1 # Copyright (c) 2012-2013 Mitch Garnaat http://garnaat.org/
      2 # Copyright 2012-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
      3 #
   (...)
     12 # ANY KIND, either express or implied. See the License for the specific
     13 # language governing permissions and limitations under the License.
     14 import copy
---> 16 from botocore.utils import merge_dicts
     19 def build_retry_config(
     20     endpoint_prefix, retry_model, definitions, client_retry_config=None
     21 ):
     22     service_config = retry_model.get(endpoint_prefix, {})

File /srv/conda/envs/notebook/lib/python3.10/site-packages/botocore/utils.py:34
     32 import botocore
     33 import botocore.awsrequest
---> 34 import botocore.httpsession
     36 # IP Regexes retained for backwards compatibility
     37 from botocore.compat import HEX_PAT  # noqa: F401

File /srv/conda/envs/notebook/lib/python3.10/site-packages/botocore/httpsession.py:41
     37     from ssl import OP_NO_TICKET, PROTOCOL_TLS_CLIENT
     39 try:
     40     # Always import the original SSLContext, even if it has been patched
---> 41     from urllib3.contrib.pyopenssl import orig_util_SSLContext as SSLContext
     42 except ImportError:
     43     from urllib3.util.ssl_ import SSLContext

File /srv/conda/envs/notebook/lib/python3.10/site-packages/urllib3/contrib/pyopenssl.py:50
      1 """
      2 TLS with SNI_-support for Python 2. Follow these instructions if you would
      3 like to verify TLS certificates in Python 2. Note, the default libraries do
   (...)
     46 .. _idna: https://github.com/kjd/idna
     47 """
     48 from __future__ import absolute_import
---> 50 import OpenSSL.SSL
     51 from cryptography import x509
     52 from cryptography.hazmat.backends.openssl import backend as openssl_backend

File /srv/conda/envs/notebook/lib/python3.10/site-packages/OpenSSL/__init__.py:8
      1 # Copyright (C) AB Strakt
      2 # See LICENSE for details.
      4 """
      5 pyOpenSSL - A simple wrapper around the OpenSSL library
      6 """
----> 8 from OpenSSL import crypto, SSL
      9 from OpenSSL.version import (
     10     __author__,
     11     __copyright__,
   (...)
     17     __version__,
     18 )
     21 __all__ = [
     22     "SSL",
     23     "crypto",
   (...)
     31     "__version__",
     32 ]

File /srv/conda/envs/notebook/lib/python3.10/site-packages/OpenSSL/crypto.py:3279
   3259 load_pkcs12 = utils.deprecated(
   3260     load_pkcs12,
   3261     __name__,
   (...)
   3266     DeprecationWarning,
   3267 )
   3270 # There are no direct unit tests for this initialization.  It is tested
   3271 # indirectly since it is necessary for functions like dump_privatekey when
   3272 # using encryption.
   (...)
   3277 # OpenSSL library (and is linked against the same one that cryptography is
   3278 # using)).
-> 3279 _lib.OpenSSL_add_all_algorithms()
   3281 # This is similar but exercised mainly by exception_from_error_queue.  It calls
   3282 # both ERR_load_crypto_strings() and ERR_load_SSL_strings().
   3283 _lib.SSL_load_error_strings()

AttributeError: module 'lib' has no attribute 'OpenSSL_add_all_algorithms'
```

**Comment by brian-rose** [comment]
@droumis would you be able to take a look at this?

**Comment by droumis** [comment]
solved by #12
